### PR TITLE
Histogram changes split out from master

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
@Moelf Since there were some issues with these changes (https://github.com/JuliaPlots/StatsPlots.jl/issues/357, https://github.com/JuliaPlots/StatsPlots.jl/issues/355) I split these out to this branch.

### CODE REVIEW

**Comment by Moelf** [comment]
thx, I will take a look at a look today see if I can fix them

**Comment by Moelf** [comment]
ok, so #349 is broken again, let me try again, sorry for the trouble

**Comment by Moelf** [comment]
from what I can tell, revert this does **not** fix:
https://github.com/JuliaPlots/StatsPlots.jl/issues/355#issuecomment-662153378
nor
https://github.com/JuliaPlots/StatsPlots.jl/issues/355#issuecomment-662139562

and it breaks #343, am I missing something?

**Comment by BeastyBlacksmith** [comment]
closing in favor of #365
